src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/StreamExtensions.NetCore.cs

```diff
@@ -90,7 +97,7 @@ public static async Task WriteProjectInfoAsync(this Stream stream, RazorProjectI
         // The array may be larger than the bytes read so make sure to trim accordingly.
         var projectInfoMemory = projectInfoBytes.AsMemory(0, sizeToRead);
 
-        return RazorProjectInfo.DeserializeFrom(projectInfoMemory);
+        return RazorMessagePackSerializer.DeserializeFrom<RazorProjectInfo>(projectInfoMemory);
```

Comment:
This might not be something you're planning for this change, but consider calling the `DeserializeFromAsync` method you added above that takes a stream.

```C#
return RazorMessagePackSerializer.DeserializeFromAsync<RazorProjectInfo>(stream, cancellation);
```